packages/knip/test/cli/reporter/cli-reporter-codeclimate4.test.ts

```diff
@@ -0,0 +1,64 @@
+import { test } from 'bun:test';
+import assert from 'node:assert/strict';
+import { resolve } from '../../../src/util/path.js';
+import { assertAndRemoveProperty } from '../../helpers/assertAndRemoveProperty.js';
+import { execFactory } from '../../helpers/exec.js';
+
+const cwd = resolve('fixtures/dependencies');
+
+const exec = execFactory(cwd);
+
+test('knip --reporter codeclimate (dependencies)', () => {
+  const json = [
+    {
+      type: 'issue',
+      check_name: 'Unused dependencies',
+      description: '@tootallnate/once',
+      categories: ['Bug Risk'],
+      location: { path: 'package.json', positions: { begin: { line: 8, column: 6 } } },
+      severity: 'major',
+    },
+    {
+      type: 'issue',
+      check_name: 'Unused dependencies',
+      description: 'fs-extra',
+      categories: ['Bug Risk'],
+      location: { path: 'package.json', positions: { begin: { line: 10, column: 6 } } },
+      severity: 'major',
+    },
+    {
+      type: 'issue',
+      check_name: 'Unused devDependencies',
+      description: 'mocha',
+      categories: ['Bug Risk'],
+      location: { path: 'package.json', positions: { begin: { line: 23, column: 6 } } },
+      severity: 'major',
+    },
+    {
+      type: 'issue',
+      check_name: 'Unlisted binaries',
+      description: 'jest',
+      categories: ['Bug Risk'],
+      location: { path: 'package.json', positions: { begin: { line: 0, column: 0 } } },
+      severity: 'major',
+    },
+    {
+      type: 'issue',
+      check_name: 'Unlisted binaries',
+      description: 'start-server',
+      categories: ['Bug Risk'],
+      location: { path: 'package.json', positions: { begin: { line: 0, column: 0 } } },
```

Comment:
I think it's better to not add `positions` if we don't have those, otherwise it could be confusing (like here). Looks like the spec allows to omit it?

Reply:
That's true. I'll fix this.

Reply:
Hm, the GitLab documentation says that `positions` or `lines` properties are required. In the Codeclimate specs I can't find anything either that it's optional.

When there is no column, I use the `lines` property instead of `positions`.

Reply:
Thank you!

https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#locations:

Here it is mentioned that `path` is required, not `positions`. But indeed from the GitLab spec it sounds like it's required. Bummer. Let's roll with it 🛼